core/tests/protractor_utils/PreferencesPage.js

```diff
@@ -41,13 +41,19 @@ var PreferencesPage = function() {
     by.css('.protractor-test-creator-dashboard-radio'));
   var learnerDashboardRadio = element(
     by.css('.protractor-test-learner-dashboard-radio'));
+  var subInterest = element(
+    by.css('select2-search__field'));
```

Comment:
Actually, let's change the HTML of the preferences page to add a class that follows the pre-existing naming pattern: `by.css('.protractor-test-creator-dashboard-subinterest-input')`.

Reply:
I too was thinking to do that. I have looked into `preferences-page` for the changes, but I couldn't find the HTML file for the page. Can I get the location of the HTML file where I can make the desired change?

Reply:
I guess this is the file that you need: https://github.com/oppia/oppia/blob/develop/core/templates/dev/head/pages/preferences-page/preferences-page.directive.html

Reply:
Could you please have a look as to how preferred audio language is added [here](https://github.com/oppia/oppia/pull/7456/files)? Also, the html file for preferences page is the one that I've already shared. (I also don't think the file name here could be changed to something `protractor-test-creator-dashboard-subinterest-input` looking at https://github.com/oppia/oppia/blob/develop/core/tests/protractor_utils/PreferencesPage.js#L42 and few others above it)

Also, please resolve the merge conflicts and feel free to let me know if you need any help. Thanks ~

Reply:
@bansalnitish, thanks for sharing the file. 
I have already gone through the tests for `system language` and `audio language` and as well as inspected the preferences page of the oppia. 
What I found was the languages test has the options for the languages already written, like options for languages- English, Hindi etc are already written in some file and so for the audio language also.
What I cannot find is the codebase for those functions. If I can get the code for the options of  the languages, then I can get some idea of how to put the options and can conduct the test for subject interests. Can you give me any idea of where to find the code for the `options` for the languages? 
I am a very new contributor on github and I have started learning AngularJs while contributing to this organization. Can you help me?

Reply:
Hi, The languages for supported site language is here -https://github.com/oppia/oppia/blob/develop/assets/constants.ts#L230. (The audio languages are also mentioned in the same file just below this)

I'm glad to hear you're learning AngularJS and also I'm more than happy to help you :). Please let me know whenever you need. Thanks!

Reply:
Thank you @bansalnitish for your quick response and your kind help. I will let you know whatever will need to resolve this issue :)

Reply:
Awesome!